### DISCUSSION TITLE
Monorepos vs. polyliths

### DISCUSSION THREAD

**Discussion by emmeowzing** [relaxdiego/relaxdiego.github.com#36] [General]
I liked your post on monorepos and the tradeoffs between microservices and monorepos when it comes to code reuse. I've also found it difficult to share code and have used the pattern of splitting code into libraries that need to be packaged up and published to a public or private registry for reuse.

In my short time as an SRE/DevOps engineer, I've implemented a couple patterns for automation supporting monorepos and I've really grown to appreciate the simplicity they offer as far as code reuse. As of late I had a thought about sharing code more generally throughout a company's broader codebase, however, and I came across the polylith-style of code organization. Specifically,

https://polylith.gitbook.io/polylith

I'm interested in eventually giving it a try. What do you think of this approach?

**Comment by relaxdiego** [comment]
Thanks for your comments, @emmeowzing! Polylith sounds familiar and I may have come across it in the past. I like its promise of better code re-usability without the self-imposed friction. What I'm concerned with about its approach is that the repository's structure appears to be dictating the software's architecture.

Example, in our team's case, we stick with Alistair Cockburn's Hexagonal Architecture since it provides very clear guidance on what belongs in or outside the domain logic and results in very maintainable code. Polylith's prescription seems to conflict with it. I could be wrong though (admittedly, I've only taken a cursory look at the Polylith docs) so I'm interested on what you think about that.
